### ISSUE TITLE
Consistent JVM exceptions in large games

### ISSUE THREAD

**Issue by p996** [vjurenka/BWMirror#8]
It seems that whenever a game reaches a large number of units the JVM will at some point crash if the bwapi.Game.canBuildHere(JLbwapi/TilePosition;Lbwapi/UnitType;Lbwapi/Unit;) function is called.  My code checks to make sure that all parameters passed to the function are valid.  None of the parameters are null, the Unit exists, and the TilePosition is within the map boundaries.  This only happens in large games such as a 3 vs 3 or 4 vs 4 with usually 500+ units existing on the map and potentially into the thousands which have existed at some point in the game.  My bot did not have this issue with BWAPI 3.7.4 version of BWMirror so I suspect it may be a BWAPI 4 issue, though, I'm not sure.  The most recent crash occurred when the function was passed a TilePosition at (116,6) (The map is 256x256), UnitType.Zerg_Creep_Colony, and a Zerg_Drone with unit ID 1497.  I can provide more information, code, or the map if it would be helpful.

Here is the JVM log from the most recent crash:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x6fb8196c, pid=3208, tid=8464
#
# JRE version: Java(TM) SE Runtime Environment (8.0_40-b26) (build 1.8.0_40-b26)
# Java VM: Java HotSpot(TM) Client VM (25.40-b25 mixed mode windows-x86 )
# Problematic frame:
# C  [bwapi_bridge2_3.dll+0x3196c]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  T H R E A D  ---------------

Current thread (0x00dcc400):  JavaThread "main" [_thread_in_native, id=8464, stack(0x00ee0000,0x00f30000)]

siginfo: ExceptionCode=0xc0000005, reading address 0x00000010

Registers:
EAX=0x00000000, EBX=0x00000e80, ECX=0x00f2d4cc, EDX=0x00000000
ESP=0x00f2d404, EBP=0x00f2d52c, ESI=0x00f2d49c, EDI=0x00000000
EIP=0x6fb8196c, EFLAGS=0x00010246

Top of Stack: (sp=0x00f2d404)
0x00f2d404:   00000000 54fd4cbb 6fd3633c 00f2d650
0x00f2d414:   00f2d5cc 00000074 00f2d4a0 48090338
0x00f2d424:   00e0c598 00000074 00000000 00dcc400
0x00f2d434:   00f2d420 00f2d5bc 00000006 00000076
0x00f2d444:   00000ec0 00000006 00f2d55c 00000e80
0x00f2d454:   000000c0 47c10260 00000001 003e5078
0x00f2d464:   003e50b8 003e50b8 00000007 00000008
0x00f2d474:   3f800000 000000f7 00000100 00000000 

Instructions: (pc=0x6fb8196c)
0x6fb8194c:   00 00 15 8b 8c 24 d4 00 00 00 85 c9 74 20 8b 11
0x6fb8195c:   8d 84 24 c4 00 00 00 3b c8 0f 95 c0 0f b6 c0 50
0x6fb8196c:   ff 52 10 c7 84 24 d4 00 00 00 00 00 00 00 c6 84
0x6fb8197c:   24 ac 01 00 00 16 8b 8c 24 04 01 00 00 85 c9 74 


Register to memory mapping:

EAX=0x00000000 is an unknown value
EBX=0x00000e80 is an unknown value
ECX=0x00f2d4cc is pointing into the stack for thread: 0x00dcc400
EDX=0x00000000 is an unknown value
ESP=0x00f2d404 is pointing into the stack for thread: 0x00dcc400
EBP=0x00f2d52c is pointing into the stack for thread: 0x00dcc400
ESI=0x00f2d49c is pointing into the stack for thread: 0x00dcc400
EDI=0x00000000 is an unknown value


Stack: [0x00ee0000,0x00f30000],  sp=0x00f2d404,  free space=309k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [bwapi_bridge2_3.dll+0x3196c]
C  [bwapi_bridge2_3.dll+0x1ea370]

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  bwapi.Game.canBuildHere_native(JLbwapi/TilePosition;Lbwapi/UnitType;Lbwapi/Unit;)Z+0
j  bwapi.Game.canBuildHere(Lbwapi/TilePosition;Lbwapi/UnitType;Lbwapi/Unit;)Z+8
j  production.Builder.canBuildHere(Lproduction/Tile;)Z+393
J 324 C1 production.Builder.getNear(Lproduction/Tile;)Lproduction/Tile; (121 bytes) @ 0x02853518 [0x02853420+0xf8]
j  production.Builder.getMorphTile(Lproduction/Tile;)Lproduction/Tile;+138
j  production.Builder.<init>(Levent/Base;Lbwapi/Unit;Lproduction/BuildRequest;Lproduction/Tile;)V+120
J 342 C1 production.BuildRequest.requestBuild(Levent/Base;)Z (2131 bytes) @ 0x0285aa94 [0x02858bb0+0x1ee4]
j  composition.ExpansionSetup.buildNext(Levent/Base;)V+13
J 199 C1 event.Base.onFrame()V (141 bytes) @ 0x0282664c [0x02826440+0x20c]
J 202 C1 event.StarcraftListener.onFrame()V (290 bytes) @ 0x02828664 [0x028280f0+0x574]
J 215 C1 bwapi.AIModule.onFrame()V (17 bytes) @ 0x0282de08 [0x0282ddd0+0x38]
v  ~StubRoutines::call_stub
j  bwapi.Mirror.startGame()V+0
j  event.StarcraftListener.run()V+15
j  event.StarcraftListener.main([Ljava/lang/String;)V+7
v  ~StubRoutines::call_stub

---------------  P R O C E S S  ---------------

Java Threads: ( => current thread )
  0x45053800 JavaThread "Service Thread" daemon [_thread_blocked, id=7184, stack(0x45bb0000,0x45c00000)]
  0x4503dc00 JavaThread "C1 CompilerThread0" daemon [_thread_blocked, id=6732, stack(0x45ab0000,0x45b00000)]
  0x45035c00 JavaThread "JDWP Command Reader" daemon [_thread_in_native, id=7152, stack(0x459e0000,0x45a30000)]
  0x45034c00 JavaThread "JDWP Event Helper Thread" daemon [_thread_blocked, id=8884, stack(0x45830000,0x45880000)]
  0x45033800 JavaThread "JDWP Transport Listener: dt_socket" daemon [_thread_blocked, id=6352, stack(0x45340000,0x45390000)]
  0x45025400 JavaThread "Attach Listener" daemon [_thread_blocked, id=6952, stack(0x452f0000,0x45340000)]
  0x4502d800 JavaThread "Signal Dispatcher" daemon [_thread_blocked, id=4100, stack(0x00d60000,0x00db0000)]
  0x44ff4400 JavaThread "Finalizer" daemon [_thread_blocked, id=7612, stack(0x45260000,0x452b0000)]
  0x44fee000 JavaThread "Reference Handler" daemon [_thread_blocked, id=8872, stack(0x04760000,0x047b0000)]
=>0x00dcc400 JavaThread "main" [_thread_in_native, id=8464, stack(0x00ee0000,0x00f30000)]

Other Threads:
  0x44fea400 VMThread [stack: 0x00c50000,0x00ca0000] [id=8800]
  0x45055000 WatcherThread [stack: 0x45890000,0x458e0000] [id=7284]

VM state:not at safepoint (normal execution)

VM Mutex/Monitor currently owned by a thread: None

Heap:
 def new generation   total 4928K, used 2514K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K,  56% used [0x04800000, 0x04a6bde8, 0x04c50000)
  from space 512K,   6% used [0x04cd0000, 0x04cd8b70, 0x04d50000)
  to   space 512K,   0% used [0x04c50000, 0x04c50000, 0x04cd0000)
 tenured generation   total 10944K, used 3636K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dd388, 0x1a0dd400, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K

Card table byte_map: [0x44800000,0x44a10000] byte_map_base: 0x447dc000

Polling page: 0x00170000

CodeCache: size=32768Kb used=1308Kb max_used=1308Kb free=31459Kb
 bounds [0x02740000, 0x02888000, 0x04740000]
 total_blobs=561 nmethods=419 adapters=81
 compilation: enabled

Compilation events (10 events):
Event: 588.941 Thread 0x4503dc00  414             composition.Necessities::getPriorityRequest (430 bytes)
Event: 588.944 Thread 0x4503dc00 nmethod 414 0x028829c8 code [0x02882d70, 0x028841d0]
Event: 620.474 Thread 0x4503dc00  415             combat.AirInvasionBrigade::readyToAttack (97 bytes)
Event: 620.475 Thread 0x4503dc00 nmethod 415 0x02885b48 code [0x02885c70, 0x02885ec8]
Event: 635.856 Thread 0x4503dc00  416             bwapi.WeaponType::get (53 bytes)
Event: 635.856 Thread 0x4503dc00 nmethod 416 0x02886148 code [0x02886280, 0x02886430]
Event: 650.456 Thread 0x4503dc00  418             composition.Necessities::isComplete (12 bytes)
Event: 650.457 Thread 0x4503dc00 nmethod 418 0x028867c8 code [0x028868d0, 0x0288698c]
Event: 714.124 Thread 0x4503dc00  421             event.StrategySelector::hasNecessities (46 bytes)
Event: 714.124 Thread 0x4503dc00 nmethod 421 0x02886f08 code [0x02887020, 0x02887158]

GC Heap History (10 events):
Event: 832.495 GC heap before
{Heap before GC invocations=1018 (full 0):
 def new generation   total 4928K, used 4445K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K, 100% used [0x04800000, 0x04c50000, 0x04c50000)
  from space 512K,   5% used [0x04c50000, 0x04c57520, 0x04cd0000)
  to   space 512K,   0% used [0x04cd0000, 0x04cd0000, 0x04d50000)
 tenured generation   total 10944K, used 3627K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dafe8, 0x1a0db000, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
Event: 832.495 GC heap after
Heap after GC invocations=1019 (full 0):
 def new generation   total 4928K, used 30K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K,   0% used [0x04800000, 0x04800000, 0x04c50000)
  from space 512K,   5% used [0x04cd0000, 0x04cd79f8, 0x04d50000)
  to   space 512K,   0% used [0x04c50000, 0x04c50000, 0x04cd0000)
 tenured generation   total 10944K, used 3629K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0db750, 0x1a0db800, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
}
Event: 832.989 GC heap before
{Heap before GC invocations=1019 (full 0):
 def new generation   total 4928K, used 4446K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K, 100% used [0x04800000, 0x04c50000, 0x04c50000)
  from space 512K,   5% used [0x04cd0000, 0x04cd79f8, 0x04d50000)
  to   space 512K,   0% used [0x04c50000, 0x04c50000, 0x04cd0000)
 tenured generation   total 10944K, used 3629K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0db750, 0x1a0db800, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
Event: 832.990 GC heap after
Heap after GC invocations=1020 (full 0):
 def new generation   total 4928K, used 31K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K,   0% used [0x04800000, 0x04800000, 0x04c50000)
  from space 512K,   6% used [0x04c50000, 0x04c57fe0, 0x04cd0000)
  to   space 512K,   0% used [0x04cd0000, 0x04cd0000, 0x04d50000)
 tenured generation   total 10944K, used 3631K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dbd68, 0x1a0dbe00, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
}
Event: 833.463 GC heap before
{Heap before GC invocations=1020 (full 0):
 def new generation   total 4928K, used 4447K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K, 100% used [0x04800000, 0x04c50000, 0x04c50000)
  from space 512K,   6% used [0x04c50000, 0x04c57fe0, 0x04cd0000)
  to   space 512K,   0% used [0x04cd0000, 0x04cd0000, 0x04d50000)
 tenured generation   total 10944K, used 3631K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dbd68, 0x1a0dbe00, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
Event: 833.464 GC heap after
Heap after GC invocations=1021 (full 0):
 def new generation   total 4928K, used 35K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K,   0% used [0x04800000, 0x04800000, 0x04c50000)
  from space 512K,   6% used [0x04cd0000, 0x04cd8d50, 0x04d50000)
  to   space 512K,   0% used [0x04c50000, 0x04c50000, 0x04cd0000)
 tenured generation   total 10944K, used 3633K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dc528, 0x1a0dc600, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
}
Event: 833.908 GC heap before
{Heap before GC invocations=1021 (full 0):
 def new generation   total 4928K, used 4451K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K, 100% used [0x04800000, 0x04c50000, 0x04c50000)
  from space 512K,   6% used [0x04cd0000, 0x04cd8d50, 0x04d50000)
  to   space 512K,   0% used [0x04c50000, 0x04c50000, 0x04cd0000)
 tenured generation   total 10944K, used 3633K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dc528, 0x1a0dc600, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
Event: 833.908 GC heap after
Heap after GC invocations=1022 (full 0):
 def new generation   total 4928K, used 35K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K,   0% used [0x04800000, 0x04800000, 0x04c50000)
  from space 512K,   6% used [0x04c50000, 0x04c58ec8, 0x04cd0000)
  to   space 512K,   0% used [0x04cd0000, 0x04cd0000, 0x04d50000)
 tenured generation   total 10944K, used 3634K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dcbd8, 0x1a0dcc00, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
}
Event: 834.361 GC heap before
{Heap before GC invocations=1022 (full 0):
 def new generation   total 4928K, used 4451K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K, 100% used [0x04800000, 0x04c50000, 0x04c50000)
  from space 512K,   6% used [0x04c50000, 0x04c58ec8, 0x04cd0000)
  to   space 512K,   0% used [0x04cd0000, 0x04cd0000, 0x04d50000)
 tenured generation   total 10944K, used 3634K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dcbd8, 0x1a0dcc00, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
Event: 834.362 GC heap after
Heap after GC invocations=1023 (full 0):
 def new generation   total 4928K, used 34K [0x04800000, 0x04d50000, 0x19d50000)
  eden space 4416K,   0% used [0x04800000, 0x04800000, 0x04c50000)
  from space 512K,   6% used [0x04cd0000, 0x04cd8b70, 0x04d50000)
  to   space 512K,   0% used [0x04c50000, 0x04c50000, 0x04cd0000)
 tenured generation   total 10944K, used 3636K [0x19d50000, 0x1a800000, 0x44800000)
   the space 10944K,  33% used [0x19d50000, 0x1a0dd388, 0x1a0dd400, 0x1a800000)
 Metaspace       used 2319K, capacity 2754K, committed 2880K, reserved 4480K
}

Deoptimization events (0 events):
No events

Internal exceptions (2 events):
Event: 0.032 Thread 0x00dcc400 Exception <a 'java/lang/NoSuchMethodError': Method sun.misc.Unsafe.defineClass(Ljava/lang/String;[BII)Ljava/lang/Class; name or signature does not match> (0x0480b6a8) thrown at [C:\re\workspace\8-2-build-windows-i586-cygwin\jdk8u40\3098\hotspot\src\share\vÛDgoG ?
Event: 0.032 Thread 0x00dcc400 Exception <a 'java/lang/NoSuchMethodError': Method sun.misc.Unsafe.prefetchRead(Ljava/lang/Object;J)V name or signature does not match> (0x0480b928) thrown at [C:\re\workspace\8-2-build-windows-i586-cygwin\jdk8u40\3098\hotspot\src\share\vm\prims\jni.cpp, l

Events (10 events):
Event: 832.495 Executing VM operation: GenCollectForAllocation
Event: 832.495 Executing VM operation: GenCollectForAllocation done
Event: 832.989 Executing VM operation: GenCollectForAllocation
Event: 832.990 Executing VM operation: GenCollectForAllocation done
Event: 833.463 Executing VM operation: GenCollectForAllocation
Event: 833.464 Executing VM operation: GenCollectForAllocation done
Event: 833.907 Executing VM operation: GenCollectForAllocation
Event: 833.908 Executing VM operation: GenCollectForAllocation done
Event: 834.361 Executing VM operation: GenCollectForAllocation
Event: 834.362 Executing VM operation: GenCollectForAllocation done


Dynamic libraries:
0x01030000 - 0x01063000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\javaw.exe
0x76e90000 - 0x77010000     C:\Windows\SysWOW64\ntdll.dll
0x748f0000 - 0x74a00000     C:\Windows\syswow64\kernel32.dll
0x768d0000 - 0x76917000     C:\Windows\syswow64\KERNELBASE.dll
0x75bc0000 - 0x75c60000     C:\Windows\syswow64\ADVAPI32.dll
0x74e30000 - 0x74edc000     C:\Windows\syswow64\msvcrt.dll
0x76a70000 - 0x76a89000     C:\Windows\SysWOW64\sechost.dll
0x75120000 - 0x75210000     C:\Windows\syswow64\RPCRT4.dll
0x74820000 - 0x74880000     C:\Windows\syswow64\SspiCli.dll
0x74810000 - 0x7481c000     C:\Windows\syswow64\CRYPTBASE.dll
0x74d30000 - 0x74e30000     C:\Windows\syswow64\USER32.dll
0x74c60000 - 0x74cf0000     C:\Windows\syswow64\GDI32.dll
0x76e60000 - 0x76e6a000     C:\Windows\syswow64\LPK.dll
0x753c0000 - 0x7545d000     C:\Windows\syswow64\USP10.dll
0x73f20000 - 0x740be000     C:\Windows\WinSxS\x86_microsoft.windows.common-controls_6595b64144ccf1df_6.0.7601.17514_none_41e6975e2bd6f2b2\COMCTL32.dll
0x769b0000 - 0x76a07000     C:\Windows\syswow64\SHLWAPI.dll
0x75800000 - 0x75860000     C:\Windows\system32\IMM32.DLL
0x75460000 - 0x7552c000     C:\Windows\syswow64\MSCTF.dll
0x70400000 - 0x704bf000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\msvcr100.dll
0x5fc50000 - 0x60010000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\client\jvm.dll
0x73b10000 - 0x73b17000     C:\Windows\system32\WSOCK32.dll
0x75a20000 - 0x75a55000     C:\Windows\syswow64\WS2_32.dll
0x74f70000 - 0x74f76000     C:\Windows\syswow64\NSI.dll
0x72280000 - 0x722b2000     C:\Windows\system32\WINMM.dll
0x72360000 - 0x72369000     C:\Windows\system32\VERSION.dll
0x748e0000 - 0x748e5000     C:\Windows\syswow64\PSAPI.DLL
0x739c0000 - 0x739cc000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\verify.dll
0x73960000 - 0x73981000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\java.dll
0x73290000 - 0x732b9000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\jdwp.dll
0x733b0000 - 0x733b7000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\npt.dll
0x73940000 - 0x73953000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\zip.dll
0x75c80000 - 0x768cb000     C:\Windows\syswow64\SHELL32.dll
0x74f90000 - 0x750ec000     C:\Windows\syswow64\ole32.dll
0x74f80000 - 0x74f8b000     C:\Windows\syswow64\profapi.dll
0x733a0000 - 0x733a8000     C:\Program Files (x86)\Java\jre1.8.0_40\bin\dt_socket.dll
0x73010000 - 0x73020000     C:\Windows\system32\NLAapi.dll
0x73000000 - 0x73010000     C:\Windows\system32\napinsp.dll
0x72fe0000 - 0x72ff2000     C:\Windows\system32\pnrpnsp.dll
0x72760000 - 0x7279c000     C:\Windows\System32\mswsock.dll
0x72710000 - 0x72754000     C:\Windows\system32\DNSAPI.dll
0x72fd0000 - 0x72fd8000     C:\Windows\System32\winrnr.dll
0x73d70000 - 0x73d8c000     C:\Windows\system32\IPHLPAPI.DLL
0x743b0000 - 0x743b7000     C:\Windows\system32\WINNSI.DLL
0x726d0000 - 0x72708000     C:\Windows\System32\fwpuclnt.dll
0x726c0000 - 0x726c6000     C:\Windows\system32\rasadhlp.dll
0x726a0000 - 0x726a5000     C:\Windows\System32\wshtcpip.dll
0x6fb50000 - 0x6fdaf000     C:\Users\human\workspace\Bot\bwapi_bridge2_3.dll
0x72510000 - 0x72581000     C:\Windows\system32\MSVCP120.dll
0x72420000 - 0x7250e000     C:\Windows\system32\MSVCR120.dll
0x6acc0000 - 0x6ad27000     C:\Windows\libgmp-10.dll
0x6f540000 - 0x6f59e000     C:\Windows\libmpfr-4.dll
0x71a80000 - 0x71b6b000     C:\Windows\system32\dbghelp.dll

VM Arguments:
jvm_args: -agentlib:jdwp=transport=dt_socket,suspend=y,address=localhost:61661 -Xmx1024M -Dfile.encoding=Cp1252 
java_command: event.StarcraftListener
java_class_path (initial): C:\Users\human\workspace\Bot\bin;C:\Users\human\workspace\Bot\lib\bwmirror_v2_3.jar
Launcher Type: SUN_STANDARD

Environment Variables:
PATH=C:/Program Files (x86)/Java/jre1.8.0_40/bin/client;C:/Program Files (x86)/Java/jre1.8.0_40/bin;C:/Program Files (x86)/Java/jre1.8.0_40/lib/i386;C:\Users\human\Desktop\;C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\ATI Technologies\ATI.ACE\Core-Static;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Users\human\Desktop\eclipsex86;
USERNAME=human
OS=Windows_NT
PROCESSOR_IDENTIFIER=AMD64 Family 16 Model 4 Stepping 2, AuthenticAMD



---------------  S Y S T E M  ---------------

OS: Windows 7 , 64 bit Build 7601 (6.1.7601.18409)

CPU:total 4 (4 cores per cpu, 1 threads per core) family 16 model 4 stepping 2, cmov, cx8, fxsr, mmx, sse, sse2, sse3, popcnt, mmxext, 3dnowpref, lzcnt, sse4a, tsc, tscinvbit, tscinv

Memory: 4k page, physical 16774684k(10288620k free), swap 16772828k(10139180k free)

vm_info: Java HotSpot(TM) Client VM (25.40-b25) for windows-x86 JRE (1.8.0_40-b26), built on Mar  7 2015 13:51:08 by "java_re" with MS VC++ 10.0 (VS2010)

time: Wed Apr 22 11:28:36 2015
elapsed time: 834 seconds (0d 0h 13m 54s)
```

**Comment by vjurenka** [comment]
Hi, 
thank you for the report, especially for all the detailed info :thumbsup:.

Will look into it.

**Comment by p996** [comment]
This issue has been resolved in BWMirror version 2.5.  It seems likely that the issue was caused by the prior version of BWAPI and not BWMirror.
